onnxruntime/core/optimizer/conv_activation_fusion.cc

```diff
@@ -19,24 +19,12 @@ namespace {
 
 #if !defined(ORT_MINIMAL_BUILD)
 namespace selectors {
+
 const Node* GetLoneConsumerNode(const GraphViewer& graph_viewer, const Node& node) {
   if (!optimizer_utils::CheckOutputEdges(graph_viewer.GetGraph(), node, 1)) {
     return nullptr;
   }
```

Comment:
Not sure this is enough but that would be an issue with CheckOutputEdges.

````
bool CheckOutputEdges(const Graph& graph, const Node& node, size_t expected_output_edges) {
  if (graph.NodeProducesGraphOutput(node)) {
    return false;
  }

  return node.GetOutputEdgesCount() == expected_output_edges;
}
```

If the output edge is to an implicit input (which it was in this case) I think all bets are off and it has to be treated as unknown how many actual edges there are. You can't fuse with something in a subgraph either.  

Most likely that combination is rare if we're looking to fuse, hence this hasn't blown up before. But I think CheckOutputEdges might need to also check that no edges go to implicit inputs and return false if any do.